questions/949-medium-anyof/template.ts

```diff
@@ -1 +1,24 @@
-export type AnyOf<T extends readonly any[]> = any
+export type AnyOf<T extends readonly any[]> = T extends [
+  infer First,
+  ...infer Rest
+]
+  ? First extends number
+    ? First extends 1
+      ? true
+      : AnyOf<Rest>
+    : First extends string
+    ? First extends ''
+      ? AnyOf<Rest>
+      : true
+    : First extends any[]
+    ? First extends []
+      ? AnyOf<Rest>
+      : true
+    : First extends Record<any, any>
+    ? keyof First extends never
+      ? AnyOf<Rest>
+      : true
+    : First extends true
+    ? true
+    : AnyOf<Rest>
+  : false
```

Comment:
I recommend not using nested conditions, because they become hard to read and unscalable. Check this alternative:
```suggestion
type Falsy = 0 | "" | false | [] | { [key: string]: never };

export type AnyOf<T extends readonly any[]> =
  T extends [infer First, ...infer Rest]
    ? First extends Falsy
      ? AnyOf<Rest>
      : true
    : false;
```